pkg/artifactory/resource/configuration/resource_artifactory_ldap_setting_v2.go

```diff
@@ -0,0 +1,504 @@
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(true),
+				Validators: []validator.Bool{
+					boolvalidator.AlsoRequires(path.Expressions{
+						path.MatchRoot("search_filter"),
+						path.MatchRoot("search_base"),
+						path.MatchRoot("manager_dn"),
+						path.MatchRoot("manager_password"),
+					}...),
+				},
+			},
+			"manager_dn": schema.StringAttribute{
+				MarkdownDescription: "The full DN of the user that binds to the LDAP server to perform user searches. Only used with `search` authentication.",
+				Optional:            true,
+				Computed:            true,
+				Default:             stringdefault.StaticString(""),
+				Validators: []validator.String{
+					stringvalidator.AlsoRequires(path.Expressions{
+						path.MatchRoot("search_filter"),
+						path.MatchRoot("search_base"),
+						path.MatchRoot("search_sub_tree"),
+						path.MatchRoot("manager_password"),
+					}...),
+				},
+			},
+			"manager_password": schema.StringAttribute{
+				MarkdownDescription: "The password of the user that binds to the LDAP server to perform the search. Only used with `search` authentication.",
+				Optional:            true,
+				Sensitive:           true,
+				Computed:            true,
+				Default:             stringdefault.StaticString(""),
+				Validators: []validator.String{
+					stringvalidator.AlsoRequires(path.Expressions{
+						path.MatchRoot("search_filter"),
+						path.MatchRoot("search_base"),
+						path.MatchRoot("search_sub_tree"),
+						path.MatchRoot("manager_dn"),
+					}...),
+				},
+			},
+		},
+	}
+}
+
+func (r *ArtifactoryLdapSettingResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+	r.ProviderData = req.ProviderData.(utilsdk.ProvderMetadata)
+}
+
+func (r *ArtifactoryLdapSettingResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *ArtifactoryLdapSettingResourceModel
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Convert from Terraform data model into API data model
+	ldapSearch := &LdapSearchAPIModel{
+		SearchFilter:  data.SearchFilter.ValueString(),
+		SearchBase:    data.SearchBase.ValueString(),
+		SearchSubTree: data.SearchSubTree.ValueBool(),
+		ManagerDn:     data.ManagerDn.ValueString(),
+	}
+	if !data.ManagerPassword.IsNull() {
+		ldapSearch.ManagerPassword = data.ManagerPassword.ValueString()
+	}
+	ldap := &ArtifactoryLdapSettingResourceAPIModel{
+		Key:                      data.Key.ValueString(),
+		Enabled:                  data.Enabled.ValueBool(),
+		LdapUrl:                  data.LdapUrl.ValueString(),
+		UserDnPattern:            data.UserDnPattern.ValueString(),
+		Search:                   *ldapSearch,
```

Comment:
The `*` here is only necessary because you declared the variable `ldapSearch` with the address of the struct. Take out the `&` and you won't need `*` here. See `Update()`.

Reply:
Thanks, forgot to remove it from `Create()`